astropy/units/quantity_helper/erfa.py

```diff
@@ -85,6 +86,12 @@ def helper_p2s(f, unit1):
     return [None], (radian, radian, unit1)
 
 
+def helper_eform(f, nounit):
+    if nounit is not None:
+        raise UnitTypeError("Ellipsoid cannot be a quantity.")
+    return [None], (None, None)
```

Comment:
Good idea to add coverage for `eform` too! But this doesn't seem right: the equatorial radius and flattening do have units: `(m, dimensionless_unscaled)` (and `from astropy.units.si import m` above)

Reply:
That was my first try... I couldn't make it work.
No matter how I changed the output dimensions, `eform` always gives me a-dimensional outputs... 
I have pushed the "dimensional" version for you to check... I'm probably missing something.

Reply:
Ah, it actually makes sense - I should have thought of this! The overrides only work if one puts in a `Quantity`as input (or output). Without that, the ufunc has no way to know a `Quantity` is expected and therefore just acts as if it regular `ndarray` input. We can actually not change this in astropy in any simple way.

In principle, the helper function will only be called if you put in a `Quantity` to hold the *output*. This is not in itself bad, but also not super useful. Perhaps the simplest right now is to just remove this again from the PR, and then in the next PR just work with the raw version, i.e., call it and then turn the output into `Quantity` there.

Reply:
> Perhaps the simplest right now is to just remove this again from the PR, and then in the next PR just work with the raw version, i.e., call it and then turn the output into Quantity there.

Thanks for your explanation! Makes sense... :)
Just to be sure: you mean I should go back to https://github.com/astropy/astropy/pull/14729/commits/34b642fa9ae3669adf411c8e9a6551775a91a1f9 or completely remove `eform` from the helpers? Thanks!